src/main/scala/devices/gpio/GPIO.scala

```diff
@@ -141,14 +144,20 @@ abstract class GPIO(busWidthBytes: Int, c: GPIOParams)(implicit p: Parameters)
     GPIOCtrlRegs.iof_sel   -> iofSelFields,
     GPIOCtrlRegs.drive     -> Seq(RegField(c.width, dsReg,
                                   RegFieldDesc("ds","Pin drive strength selection", reset=Some(0)))),
+    GPIOCtrlRegs.drive1    -> Seq(RegField(c.width, ds1Reg,
+                                  RegFieldDesc("ds1","Pin drive strength selection", reset=Some(0)))),
     GPIOCtrlRegs.input_en  -> Seq(RegField.rwReg(c.width, ieReg.io,
                                   Some(RegFieldDesc("input_en","Pin input enable", reset=Some(0))))),
     GPIOCtrlRegs.out_xor   -> Seq(RegField(c.width, xorReg,
                                   RegFieldDesc("out_xor","Output XOR (invert) enable", reset=Some(0)))),
     GPIOCtrlRegs.passthru_high_ie -> Seq(RegField(c.width, passthruHighIeReg,
                                          RegFieldDesc("passthru_high_ie", "Pass-through active-high interrupt enable", reset=Some(0)))),
     GPIOCtrlRegs.passthru_low_ie  -> Seq(RegField(c.width, passthruLowIeReg,
-                                         RegFieldDesc("passthru_low_ie", "Pass-through active-low interrupt enable", reset=Some(0))))
+                                         RegFieldDesc("passthru_low_ie", "Pass-through active-low interrupt enable", reset=Some(0)))),
+    GPIOCtrlRegs.ps        -> Seq(RegField(c.width, psReg,
+                                  RegFieldDesc("ps","Weak PU/PD Resistor Selection", reset=Some(0)))),
+    GPIOCtrlRegs.poe       -> Seq(RegField.rwReg(c.width, poeReg.io,
```

Comment:
Since not all pad types have the above signals, we should make them optional somehow, preferably by passing in an argument that describes all the fields that need to be driven, but this could be easier done with an argument to `GPIOParams`:
```
case class GPIOParams(
  address: BigInt,
  width: Int,
  includeIOF: Boolean = false,
  dsWidth: Int = 1,
  hasPSandPOE: Boolean = false) extends DeviceParams
```